source/common/quic/envoy_quic_dispatcher.cc

```diff
@@ -52,44 +54,43 @@ void EnvoyQuicDispatcher::OnConnectionClosed(quic::QuicConnectionId connection_i
   listener_stats_.downstream_cx_active_.dec();
   per_worker_stats_.downstream_cx_active_.dec();
   connection_handler_.decNumConnections();
-  quic_stat_names_.chargeQuicConnectionCloseStats(listener_config_.listenerScope(), error, source,
+  quic_stat_names_.chargeQuicConnectionCloseStats(listener_config_->listenerScope(), error, source,
                                                   /*is_upstream*/ false);
 }
 
 std::unique_ptr<quic::QuicSession> EnvoyQuicDispatcher::CreateQuicSession(
     quic::QuicConnectionId server_connection_id, const quic::QuicSocketAddress& self_address,
-    const quic::QuicSocketAddress& peer_address, absl::string_view alpn,
+    const quic::QuicSocketAddress& peer_address, absl::string_view /*alpn*/,
     const quic::ParsedQuicVersion& version, absl::string_view sni) {
   quic::QuicConfig quic_config = config();
+  // TODO(danzh) use passed-in ALPN instead of hard-coded h3 after proof source interfaces takes in
+  // ALPN.
   Network::ConnectionSocketPtr connection_socket = createServerConnectionSocket(
-      listen_socket_.ioHandle(), self_address, peer_address, std::string(sni), alpn);
+      listen_socket_.ioHandle(), self_address, peer_address, std::string(sni), "h3");
```

Comment:
What is the significance of this alpn change? Is it related to filter chain updates?

Reply:
Yes, as commented above, "h3" is hard-coded during cert lookup which requires a QUICHE change to fix. So I would like to make both lookup result consistent.

Reply:
In indeed, we only support "h3" now. Other APLN will fail the handshake.